pkg/ingester/stream.go

```diff
@@ -357,6 +368,21 @@ func (s *stream) storeEntries(ctx context.Context, entries []logproto.Entry, usa
 	return bytesAdded, storedEntries, invalid
 }
 
+func (s *stream) handleLoggingOfDuplicateEntry(entry logproto.Entry) {
+	if s.configs == nil {
+		return
+	}
+	if s.configs.LogDuplicateMetrics(s.tenant) {
+		s.metrics.duplicateLogBytesTotal.WithLabelValues(s.tenant).Add(float64(len(entry.Line)))
+	}
+	if s.configs.LogDuplicateStreamInfo(s.tenant) {
+		errMsg := fmt.Sprintf("duplicate log entry at timestamp %s for stream %s", entry.Timestamp.Format(time.RFC3339), s.labelsString)
```

Comment:
maybe you want to log the size in bytes too? (even if that's redundant with the metric above)
like `duplicate log entry with size=%d at timestamp %s for stream %s`